### DISCUSSION TITLE
Conditional Row Rules are not working

### DISCUSSION THREAD

**Discussion by eafarooqi** [Power-Components/livewire-powergrid#1804] [BUG]
The class is not added to the row. I can see the class in rules but its not actually applied to the tr

```
Rule::rows()
    ->when(fn ($commission) =>$commission->type == 'active')
     ->setAttribute('class', 'bg-danger'),
```

PG Row Attributes have rules. attributes also contain the class, but its not implemented.
```
pgRowAttributes({
    rowId: 51863,
    defaultClasses: '',
    rules: JSON.parse(`[
        {
            "forAction": "pg:rows",
            "apply": true,
            "applyLoop": false,
            "attributes": {
                "class": "bg-danger"
            },
            "disable": false,
            "hide": false,
            "toggleableVisibility": null,
            "toggleDetailView": "livewire-powergrid::components.frameworks.bootstrap5.no-toggle-detail",
            "editOnClickVisibility": null,
            "fieldHideEditOnClick": false,
            "fieldHideToggleable": false
        }
    ]`)
});
```

![image](https://github.com/user-attachments/assets/ea6959a5-88d7-4afc-afbe-ff54b1fb1e0b)

i am using bootstrap.

**Comment by Rajdev2011** [comment]
hi did you resolve this?

**Reply by eafarooqi** [reply]
no, i didn't get time to look into it. but its still not fixed. 
@luanfreitasdev can you please take a look, if its a bug. i will create an issue for this. Thanks

**Reply by luanfreitasdev** [reply]
Fixed: https://github.com/Power-Components/livewire-powergrid/releases/tag/v6.1.10

**Reply by eafarooqi** [reply]
Thanks

**Reply by arkdev1** [reply]
@luanfreitasdev hi some problem here.
I'm using a soft delete model, i'm using tailwind.

```php
Rule::rows()->when(fn($row) => $row->deleted_at != null)
->setAttribute('class', '!bg-red-200'),
```
